File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/Coverage.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Money } from "./Money";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type CoverageCostToBeneficiaryException = BackboneElement & {
    type: CodeableConcept;
    period?: Period;
};

export type CoverageCostToBeneficiary = BackboneElement & {
    type?: CodeableConcept;
    value?: Money | Quantity;
    exception?: CoverageCostToBeneficiaryException[];
};

export type CoverageClass = BackboneElement & {
    name?: string;
    type: CodeableConcept;
    value: string;
};

export type Coverage = DomainResource & {
    policyHolder?: Reference;
    beneficiary: Reference;
    contract?: Reference[];
    relationship?: CodeableConcept;
    type?: CodeableConcept;
    costToBeneficiary?: CoverageCostToBeneficiary[];
    subrogation?: boolean;
    subscriber?: Reference;
    payor: Reference[];
    status: string;
    class_?: CoverageClass[];
    identifier?: Identifier[];
    order?: number;
    network?: string;
    period?: Period;
    dependent?: string;
    subscriberId?: string;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Coverage.py
from typing import Optional, List
from base import Period
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Money
from base import Reference
from base import Identifier
from base import BackboneElement

class Coverage_CostToBeneficiary_Exception(BackboneElement):
    type: CodeableConcept
    period: Optional[Period] = None

class Coverage_CostToBeneficiary(BackboneElement):
    type: Optional[CodeableConcept] = None
    exception: Optional[List[Coverage_CostToBeneficiary_Exception]] = None
    value_money: Optional[Money] = None
    value_quantity: Optional[Quantity] = None

class Coverage_Class(BackboneElement):
    name: Optional[str] = None
    type: CodeableConcept
    value: str

class Coverage(DomainResource):
    policy_holder: Optional[Reference] = None
    beneficiary: Reference
    contract: Optional[List[Reference]] = None
    relationship: Optional[CodeableConcept] = None
    type: Optional[CodeableConcept] = None
    cost_to_beneficiary: Optional[List[Coverage_CostToBeneficiary]] = None
    subrogation: Optional[bool] = None
    subscriber: Optional[Reference] = None
    payor: list[Reference] = []
    status: str
    class_: Optional[List[Coverage_Class]] = None
    identifier: Optional[List[Identifier]] = None
    order: Optional[int] = None
    network: Optional[str] = None
    period: Optional[Period] = None
    dependent: Optional[str] = None
    subscriber_id: Optional[str] = None